### PR TITLE
[tidb-cdc] Monitor tidb table lose data, which table has one more regions.

### PR DESCRIPTION
**Pull request by GOODBOY008**
Link https://github.com/ververica/flink-cdc-connectors/issues/1159.

### CODE REVIEW

File: flink-connector-tidb-cdc/src/main/java/org/tikv/common/operation/iterator/ScanIterator.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import java.util.Iterator;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Copied from https://github.com/tikv/client-java project to fix
+ * https://github.com/tikv/client-java/issues/600 for 3.2.0 version.
+ *
+ * <p>Line 98 : Use `currentCache.size() > scanLimit` instead of `currentCache.size() > limit`.
```
Comment (leonardBang) [inline flink-connector-tidb-cdc/src/main/java/org/tikv/common/operation/iterator/ScanIterator.java:39]: minor: The line number looks wrong?

Comment (GOODBOY008) [reply flink-connector-tidb-cdc/src/main/java/org/tikv/common/operation/iterator/ScanIterator.java:39]: > minor: The line number looks wrong?

Thanks for your mind, I will fix it(line 98 is according to source code).

Comment (GOODBOY008) [reply flink-connector-tidb-cdc/src/main/java/org/tikv/common/operation/iterator/ScanIterator.java]: @leonardBang Pls cc.

**Review by leonardBang — APPROVED** [review]
Thanks @GOODBOY008 for the contribution, LGTM.

**Comment by leonardBang** [comment]
Plz feel free to merge, @GOODBOY008

**Pull request merged**